cmd/getcreds/main.go

```diff
@@ -115,84 +104,158 @@ func loadVerifyConfigFile(configFilename string) (AppConfigFile, error) {
 	return config, nil
 }
 
-func buildGetCredRequestBasicAuth(pubKeyFilename, userName string, password []byte, targetUrl string) (*http.Request, error) {
-	// parts from  https://astaxie.gitbooks.io/build-web-application-with-golang/content/en/04.5.html
+// This is now copy-paste from the server test side... probably make public and reuse.
+func createKeyBodyRequest(method, urlStr, filedata string) (*http.Request, error) {
+	//create attachment....
 	bodyBuf := &bytes.Buffer{}
 	bodyWriter := multipart.NewWriter(bodyBuf)
 
-	fileWriter, err := bodyWriter.CreateFormFile("pubkeyfile", pubKeyFilename)
+	//
+	fileWriter, err := bodyWriter.CreateFormFile("pubkeyfile", "somefilename.pub")
 	if err != nil {
+		fmt.Println("error writing to buffer")
+		//t.Fatal(err)
 		return nil, err
 	}
-	// open file handle
-	fh, err := os.Open(pubKeyFilename)
-	if err != nil {
-		return nil, err
-	}
-	defer fh.Close()
+	//when using a file this used to be: fh, err := os.Open(pubKeyFilename)
+	fh := strings.NewReader(filedata)
 
 	//iocopy
 	_, err = io.Copy(fileWriter, fh)
 	if err != nil {
-		log.Fatal(err)
+		//t.Fatal(err)
 		return nil, err
 	}
-	if *debug {
-		log.Printf("%v", bodyBuf)
-	}
 
 	contentType := bodyWriter.FormDataContentType()
 	bodyWriter.Close()
 
-	req, err := http.NewRequest("POST", targetUrl, bodyBuf)
+	// Create a request to pass to our handler. We don't have any query parameters for now, so we'll
+	// pass 'nil' as the third parameter.
+	req, err := http.NewRequest(method, urlStr, bodyBuf)
 	if err != nil {
+		//t.Fatal(err)
 		return nil, err
 	}
 	req.Header.Set("Content-Type", contentType)
 
-	req.SetBasicAuth(userName, string(password[:]))
 	return req, nil
 }
 
-func getCertFromTargetUrls(pubKeyFilename, userName string, password []byte, targetUrls []string, rootCAs *x509.CertPool) (cert []byte, err error) {
-	success := false
-	for _, baseUrl := range targetUrls {
-		targetUrl := baseUrl + userName
-		log.Printf("attempting to target '%s'", targetUrl)
-		tlsConfig := &tls.Config{RootCAs: rootCAs, MinVersion: tls.VersionTLS12}
-		tr := &http.Transport{
-			TLSClientConfig: tlsConfig,
-		}
-		client := &http.Client{Transport: tr, Timeout: time.Duration(5) * time.Second}
+func doCertRequest(client *http.Client, authCookies []*http.Cookie, url, filedata string) ([]byte, error) {
 
-		// TODO: change basic auth for some form of tokens.
-		//req, err := http.NewRequest("POST", targetUrl, bodyBuf)
-		req, err := buildGetCredRequestBasicAuth(pubKeyFilename, userName, password, targetUrl)
-		if err != nil {
-			log.Fatal(err)
-		}
+	req, err := createKeyBodyRequest("POST", url, filedata)
+	if err != nil {
+		return nil, err
+	}
+	//add the login cookies
+	for _, cookie := range authCookies {
+		req.AddCookie(cookie)
+	}
+	resp, err := client.Do(req) //client.Get(targetUrl)
+	if err != nil {
+		log.Printf("Failure to do x509 req")
```

Comment:
It's often helpful to print out the actual error as well. Same comment applies elsewhere.